### DISCUSSION TITLE
Feature (or example) request: Brush

### DISCUSSION THREAD

**Discussion by jfsiii** [plouc/nivo#2748] [Ideas]
This is a common, but often challenging, interaction. It'd be great to see how it should work in nivo.

**Comment by guzmonne** [comment]
I created a [simple line chart with a brush using `nivo` components]( https://github.com/guzmonne/nivo-with-brush). I didn't create a pull request since I had to do some hacked actions to make it work.

1. I had to add a new child component called `Brush` inside the `Line` component.
2. I added three new props to the `Line` component that are then passed to the `Brush` component. Two to set the initial `Brush` position, and a callback to call `onBrush`.
3. I had to modify the `SvgWrapper`component to provide a handler to each children, that gives them access to a `ref` of the parent `svg` component. I needed this `ref` to convert mouse coordinates to svg coordinates.

Do you approve of these API modification, or would you do things differently. I can generate a pull request from your comments if you want.

**Comment by stahlmanDesign** [comment]
@guzmonne this is great. Your description and examples of 'hacked actions' to make this work are helpful for understanding how to customize or enhance Nivo.

Here's what I was thinking of doing with a stacked bar chart. I've seen this before with D3 but wasn't sure how to bring it to Nivo:

<img width="990" alt="capture d ecran 2018-02-18 a 20 47 48" src="https://user-images.githubusercontent.com/972804/36359869-109e62da-14ed-11e8-9cc8-ad0d31613f27.png">

**Comment by guzmonne** [comment]
I believe you could accomplish it with `nivo` by extending the `Bar` components to handle `child` components, or hacking it like I did with the `Line` component.

I published a [medium article](https://medium.com/@gmonne/create-a-brush-component-using-nivo-dataviz-library-68074f3721ba) explaining how I did it, and the shortcomings of my method.

**Comment by MarcMagnin** [comment]
Great work @guzmonne!
I've just noticed this issue while I was exposition such feature in a 'zoom' related issue.
I'm thinking about the same "generic" kinda brush feature that would simply return the range (start/end) of the selected data to be able to do whatever you want with it such as this sample from highcharts: 
https://www.highcharts.com/demo/line-time-series

Having a generic simple capability to select an area by dragging dropping the mouse to act on it is a top feature.

**Comment by ghost** [comment]
> @guzmonne this is great. Your description and examples of 'hacked actions' to make this work are helpful for understanding how to customize or enhance Nivo.
> 
> Here's what I was thinking of doing with a stacked bar chart. I've seen this before with D3 but wasn't sure how to bring it to Nivo:
> 
> <img alt="capture d ecran 2018-02-18 a 20 47 48" width="990" src="https://user-images.githubusercontent.com/972804/36359869-109e62da-14ed-11e8-9cc8-ad0d31613f27.png">

can I get your code for bar chart with nivo-brush-zoom its too urgent for me please

**Comment by swest06** [comment]
Can someone explain how to do this with the responsive heat map.

**Comment by plouc** [comment]
Since this wasn't a bug, I decided to move this to a discussion.
